controllers/release/release_adapter.go

```diff
@@ -160,6 +165,37 @@ func (a *Adapter) EnsureReleasePipelineStatusIsTracked() (results.OperationResul
 	return results.ContinueProcessing()
 }
 
+// EnsureSnapshotEnvironmentBindingIsCreated is an operation that will ensure that a SnapshotEnvironmentBinding is created
+// or updated for the current Release.
+func (a *Adapter) EnsureSnapshotEnvironmentBindingIsCreated() (results.OperationResult, error) {
```

Comment:
I don't like how we call it by the full name including Application in some spots and not others. Especially with the name potentially changing to not include Application, I think we should call it by its proper CRD name all the time and never anything else